test/framework/controlplane_helpers.go

```diff
@@ -175,45 +176,44 @@ func WaitForControlPlaneToBeReady(ctx context.Context, input WaitForControlPlane
 
 // AssertControlPlaneFailureDomainsInput is the input for AssertControlPlaneFailureDomains.
 type AssertControlPlaneFailureDomainsInput struct {
-	GetLister  GetLister
-	ClusterKey client.ObjectKey
-	// ExpectedFailureDomains is required because this function cannot (easily) infer what success looks like.
-	// In theory this field is not strictly necessary and could be replaced with enough clever logic/math.
-	ExpectedFailureDomains map[string]int
+	Lister  Lister
+	Cluster *clusterv1.Cluster
 }
 
 // AssertControlPlaneFailureDomains will look at all control plane machines and see what failure domains they were
 // placed in. If machines were placed in unexpected or wrong failure domains the expectation will fail.
 func AssertControlPlaneFailureDomains(ctx context.Context, input AssertControlPlaneFailureDomainsInput) {
-	failureDomainCounts := map[string]int{}
-
-	// Look up the cluster object to find all known failure domains.
-	cluster := &clusterv1.Cluster{}
-	Expect(input.GetLister.Get(ctx, input.ClusterKey, cluster)).To(Succeed())
-
-	for fd := range cluster.Status.FailureDomains {
-		failureDomainCounts[fd] = 0
+	Expect(ctx).NotTo(BeNil(), "ctx is required for AssertControlPlaneFailureDomains")
+	Expect(input.Lister).ToNot(BeNil(), "Invalid argument. input.Lister can't be nil when calling AssertControlPlaneFailureDomains")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling AssertControlPlaneFailureDomains")
+
+	By("Checking all the control plane machines are in the expected failure domains")
+	failureDomainCounts := map[string]struct{}{}
```

Comment:
Should this now be a string set called controlPlaneFailureDomains? (looks like we don't count anymore)

Reply:
Done!